include/Gaffer/MetadataAlgo.h

```diff
@@ -93,6 +93,7 @@ GAFFER_API bool getReadOnly( const GraphComponent *graphComponent );
 /// \undoable
 GAFFER_API void setChildNodesAreReadOnly( Node *node, bool readOnly, bool persistent = true );
 GAFFER_API bool getChildNodesAreReadOnly( const Node *node );
+GAFFER_API bool ancestorChildNodesAreReadOnly( const GraphComponent *graphComponent );
```

Comment:
This feels like a slightly awkward concept to enshrine in the API, and since we only have one use case I think it might be a bit premature to do so. Perhaps a second use case might prefer something more like `readOnlyReason()` where we return the culprit rather than a boolean? And maybe there are variations we haven't anticipated. Let's just keep this internal for now.

Reply:
Addressed here: d10715e9b0e3302ad0b941713b9bf7b96354982e+